certora/specs/AllocationVaultV2.spec

```diff
@@ -6,36 +6,46 @@ methods {
     function allocation(bytes32) external returns uint256 envfree;
 }
 
-// Check that the only functions able to change allocations are deposit, mint, withdraw, redeem, allocation, deallocate, forceDeallocate.
+//E RUN : https://prover.certora.com/output/7508195/e366be539f21406aac7fded3fbf53a5c/?anonymousKey=bfb434d7be5020b47cc637f24dcbc70f767c8138
+
+// Definition: Functions that are allowed to change allocations
+definition canChangeAllocation(method f) returns bool =
+    f.selector == sig:deposit(uint256,address).selector ||
+    f.selector == sig:mint(uint256,address).selector ||
+    f.selector == sig:withdraw(uint256,address,address).selector ||
+    f.selector == sig:redeem(uint256,address,address).selector ||
+    f.selector == sig:allocate(address,bytes,uint256).selector ||
+    f.selector == sig:deallocate(address,bytes,uint256).selector ||
+    f.selector == sig:forceDeallocate(address,bytes,uint256,address).selector;
+
+// Check that the only functions able to change allocations are deposit, mint, withdraw, redeem, allocate, deallocate, forceDeallocate.
 rule functionsChangingAllocation(env e, method f, calldataarg args) filtered {
-    f -> !f.isView &&
-         f.selector != sig:deposit(uint,address).selector &&
-         f.selector != sig:mint(uint,address).selector &&
-         f.selector != sig:withdraw(uint,address,address).selector &&
-         f.selector != sig:redeem(uint,address,address).selector &&
-         f.selector != sig:allocate(address,bytes,uint).selector &&
-         f.selector != sig:deallocate(address,bytes,uint).selector &&
-         f.selector != sig:forceDeallocate(address,bytes,uint,address).selector
+    f -> !f.isView && !canChangeAllocation(f)
 } {
     bytes32 id;
     uint256 allocationPre = allocation(id);
 
     f(e, args);
 
-    assert allocation(id) == allocationPre;
+    assert allocation(id) == allocationPre, "Only authorized functions should change allocations";
```

Comment:
I don't feel like it adds much in the case of `assert`s, because they are already documented at the top of the rule